web/backend/src/controllers/dela.ts

```diff
@@ -263,6 +238,30 @@ delaRouter.use('/*', (req, res) => {
   }
 
   const bodyData = req.body;
+
+  // special case for voting
+  const match = req.baseUrl.match('/api/evoting/forms/(.*)/vote');
+  if (match) {
+    if (!req.session.userId) {
+      res.status(401).send('Authentication required!');
+      return;
+    }
```

Comment:
this will never be called based on the code in line 235-237. 
That error should probably return a 401, like you've done here.